app/controllers/git_api/concept_exercises_controller.rb

```diff
@@ -20,29 +20,26 @@ def create
 
       uuid = SecureRandom.uuid.gsub('-', '')
       tmp_path = Pathname.new("/tmp/#{uuid}")
-      branch_name = "#{params[:track_slug]}/#{params[:exercise_slug]}-#{uuid}"
 
       `git clone https://#{Rails.application.secrets.exercism_bot_credentials}@github.com/exercism-bot/v3.git #{tmp_path}`
 
       Dir.chdir(tmp_path) do
         `git remote add upstream https://github.com/exercism/v3.git`
-        `git fetch upstream master`
+        `git fetch upstream`
         `git checkout --no-track -b #{branch_name} upstream/master`
-      end
 
-      exercise_path = tmp_path / "languages" / params[:track_slug] / "exercises" / "concept" / params[:exercise_slug]
-      exercise_docs_path = exercise_path / ".docs"
-      exercise_meta_path = exercise_path / ".meta"
+        exercise_path = tmp_path / "languages" / params[:track_slug] / "exercises" / "concept" / params[:exercise_slug]
+        exercise_docs_path = exercise_path / ".docs"
+        exercise_meta_path = exercise_path / ".meta"
 
-      Dir.mkdir(exercise_path)
-      Dir.mkdir(exercise_docs_path)
-      Dir.mkdir(exercise_meta_path)
+        Dir.mkdir(exercise_path)
+        Dir.mkdir(exercise_docs_path)
+        Dir.mkdir(exercise_meta_path)
 
-      File.open(exercise_docs_path / "instructions.md", "w") {|f|f.write params[:instructions_markdown] }
-      File.open(exercise_meta_path / "design.md", "w") {|f|f.write params[:design_markdown] }
-      File.open(exercise_path / params[:example_filename], "w") {|f|f.write params[:example_code] }
+        File.open(exercise_docs_path / "instructions.md", "w") {|f|f.write params[:instructions_markdown] }
+        File.open(exercise_meta_path / "design.md", "w") {|f|f.write params[:design_markdown] }
+        File.open(exercise_path / params[:example_filename], "w") {|f|f.write params[:example_code] }
```

Comment:
I've moved all code in a single `Dir.chdir` block to make it more clear that they are all working on the same directory